### ISSUE TITLE
dependency loop with /sysroot mount fails

### ISSUE THREAD

**Issue by keszybz** [dracutdevs/dracut#259]
```
Jul 28 02:50:18 fedora-uefi systemd[1]: Mounting /sysroot...
Jul 28 02:50:18 fedora-uefi kernel: BTRFS info (device sda4): disk space caching is enabled
Jul 28 02:50:18 fedora-uefi kernel: BTRFS info (device sda4): has skinny extents
Jul 28 02:50:18 fedora-uefi kernel: BTRFS critical (device sda4): invalid dir item name len: 23
Jul 28 02:50:18 fedora-uefi kernel: BTRFS: error (device sda4) in btrfs_replay_log:2475: errno=-5 IO failure (Failed to recover log tree)
Jul 28 02:50:18 fedora-uefi kernel: BTRFS info (device sda4): delayed_refs has NO entry
Jul 28 02:50:18 fedora-uefi kernel: BTRFS error (device sda4): cleaner transaction attach returned -30
Jul 28 02:50:18 fedora-uefi kernel: BTRFS error (device sda4): open_ctree failed
Jul 28 02:50:18 fedora-uefi systemd[1]: sysroot.mount: Mount process exited, code=exited status=32
Jul 28 02:50:18 fedora-uefi kernel: mount (328) used greatest stack depth: 10528 bytes left
Jul 28 02:50:18 fedora-uefi systemd[1]: Failed to mount /sysroot.
Jul 28 02:50:18 fedora-uefi systemd[1]: Dependency failed for Initrd Root File System.
Jul 28 02:50:18 fedora-uefi systemd[1]: Dependency failed for Reload Configuration from the Real Root.
Jul 28 02:50:18 fedora-uefi systemd[1]: initrd-parse-etc.service: Job initrd-parse-etc.service/start failed with result 'dependency'.
Jul 28 02:50:18 fedora-uefi systemd[1]: initrd-parse-etc.service: Triggering OnFailure= dependencies.
Jul 28 02:50:18 fedora-uefi systemd[1]: basic.target: Found ordering cycle on basic.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: basic.target: Found dependency on sysinit.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: basic.target: Found dependency on emergency.target/start
Jul 28 02:50:18 fedora-uefi systemd[1]: basic.target: Found dependency on dracut-pre-pivot.service/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: basic.target: Found dependency on initrd.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: basic.target: Found dependency on basic.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: basic.target: Breaking ordering cycle by deleting job sysinit.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: sysinit.target: Job sysinit.target/stop deleted to break ordering cycle starting with basic.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: initrd-root-fs.target: Job initrd-root-fs.target/start failed with result 'dependency'.
Jul 28 02:50:18 fedora-uefi systemd[1]: initrd-root-fs.target: Triggering OnFailure= dependencies.
Jul 28 02:50:18 fedora-uefi systemd[1]: emergency.target: Found ordering cycle on emergency.target/start
Jul 28 02:50:18 fedora-uefi systemd[1]: emergency.target: Found dependency on dracut-pre-pivot.service/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: emergency.target: Found dependency on initrd.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: emergency.target: Found dependency on basic.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: emergency.target: Found dependency on sysinit.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: emergency.target: Found dependency on emergency.target/start
Jul 28 02:50:18 fedora-uefi systemd[1]: emergency.target: Breaking ordering cycle by deleting job initrd.target/stop
Jul 28 02:50:18 fedora-uefi systemd[1]: initrd.target: Job initrd.target/stop deleted to break ordering cycle starting with emergency.target/start
Jul 28 02:50:18 fedora-uefi systemd[1]: sysroot.mount: Unit entered failed state.
Jul 28 02:50:18 fedora-uefi systemd[1]: Reached target Initrd File Systems.
Jul 28 02:50:18 fedora-uefi systemd[1]: Reached target Initrd Default Target.
Jul 28 02:50:18 fedora-uefi systemd[1]: Starting Setup Virtual Console...
Jul 28 02:50:18 fedora-uefi systemd[1]: Stopped dracut initqueue hook.
Jul 28 02:50:18 fedora-uefi systemd[1]: Stopped dracut pre-udev hook.
Jul 28 02:50:18 fedora-uefi systemd[1]: Stopped dracut cmdline hook.
Jul 28 02:50:18 fedora-uefi systemd[1]: Started Setup Virtual Console.
Jul 28 02:50:18 fedora-uefi systemd[1]: Starting Emergency Shell...
```

**Comment by keszybz** [comment]
Hm, and if I press ^D to exit from emergency shell, I see "Failed to start default.target: transaction is destructive", and it hangs (ctrl-alt-del reboots). Yikes.

**Comment by haraldh** [comment]
Should be fixed with commit f24d205537b094939379440ee013cca88c7582ac

**Comment by keszybz** [comment]
Cool! Thanks.
